src/prefect/tasks/monte_carlo/client.py

```diff
@@ -0,0 +1,256 @@
+        response = requests.post(
+            url=self._api_url,
+            json=dict(query=query, variables=variables),
+            headers={
+                "x-mcd-id": self.api_key_id,
+                "x-mcd-token": self.api_token,
+                "Content-Type": "application/json",
+            },
+        )
+        response.raise_for_status()
+        response = json.loads(response.text)
+        self.logger.info("Response: %s", response)
+        return response
+
+    def get_warehouse_id_by_name(self, warehouse_name: str) -> str:
+        response = self.graphql(
+            query="""
+                    {
+                        getUser {
+                            account {
+                                warehouses {
+                                    name
+                                    uuid
+                                }
+                            }
+                        }
+                    }
+                    """
+        )
+        warehouses = response["data"]["getUser"]["account"]["warehouses"]
+        id_ = [w for w in warehouses if w["name"] == warehouse_name][0].get("uuid")
+        return id_
+
+    def get_resources(self) -> List[Dict[str, Any]]:
+        response = self.graphql(
+            query="""
+                    query {
+                        getResources {
+                            name
+                            type
+                            id
+                            uuid
+                            isDefault
+                            isUserProvided
+                        }
+                    }
+                    """
+        )
+        return response["data"]["getResources"]
+
+    def get_table_mcon(self, full_table_id: str, warehouse_name: str) -> str:
+        response = self.graphql(
+            query="""
+                    query($dwh_id: UUID!, $full_table_id: String!) {
+                        getTable(dwId: $dwh_id,
+                            fullTableId: $full_table_id) {
+                            mcon
+                        }
+                    }
+                    """,
+            variables=dict(
+                dwh_id=self.get_warehouse_id_by_name(warehouse_name),
+                full_table_id=full_table_id,
+            ),
+        )
+        return response["data"]["getTable"]["mcon"]
+
+    def get_node_mcon(self, object_id: str, object_type: str) -> str:
+        response = self.graphql(
+            query="""
+                    query($object_id: String!, $object_type: String!) {
+                        getLineageNode(objectId: $object_id,
+                            objectType: $object_type) {
+                            mcon
+                        }
+                    }
+                    """,
+            variables=dict(object_id=object_id, object_type=object_type),
+        )
+        mcon = response["data"]["getLineageNode"]["mcon"]
```

Comment:
I recommend raising a custom exception (e.g. `LineageNodeNotFound`) when the specified table cannot be found.

Reply:
same here: I will entirely remove this method because it's no longer necessary, this will make things cleaner 👍